Text/CustomEntitySearch/CustomEntitySearch.cs

```diff
@@ -50,38 +52,38 @@ public static class CustomEntitySearch
             }
 
             WebApiSkillResponse response = WebApiSkillHelpers.ProcessRequestRecords(skillName, requestRecords,
-                (inRecord, outRecord) => {
+                 (inRecord, outRecord) => {
                     string text = inRecord.Data["text"] as string;
                     IList<string> words = ((JArray)inRecord.Data["words"]).ToObject<List<string>>();
-                    if (words == null)
+                    Dictionary<string, string[]> synonyms = (inRecord.Data.ContainsKey("synonyms")) ? ((JContainer)inRecord.Data["synonyms"]).ToObject<Dictionary<string, string[]>>() : new Dictionary<string, string[]>();
```

Comment:
>inRecord.Data["synonyms"] [](start = 115, length = 25)

Can you add model validations to the begining of this method so that you return "nice" erorr messages if the payload is malformed?

eg: right now if you supply malformed dictionaries you'll probably get back some hard to decipher json exception

Reply:
Which payload? Currently, the request payload can not have the keys and the skill will still run using the words.json file if no words are provided by the request payload. If the words.json file does not have any words, synonyms, etc. then the outRecord will just be empty.